core/src/data/clientGeoJsonSource.cpp

```diff
@@ -73,38 +78,28 @@ void ClientGeoJsonSource::addPoint(double* _coords) {
 
 }
 
-void ClientGeoJsonSource::addLine(double* _coords, int _lineLength) {
-    
-    std::vector<geojsonvt::LonLat> line(_lineLength, { 0, 0 });
-    for (int i = 0; i < _lineLength; i++) {
-        line[i] = { _coords[2 * i], _coords[2 * i + 1] };
-    }
-    
+void ClientGeoJsonSource::addLine(const Properties& _tags, const Coordinates& _line) {
+    auto& line = reinterpret_cast<const std::vector<geojsonvt::LonLat>&>(_line);
```

Comment:
It was probably about removing constness - which is wrong in almost any case.

Only the usual danger of reinterpret_cast applies here, that one has to be sure that the struct layout of LonLat and LngLat dont change independently.